base/pkg/github.jl

```diff
@@ -12,29 +12,30 @@ const AUTH_DATA = Dict{Any,Any}(
 )
 
 function user()
-    if !success(`git config --global github.user`)
-        error("""
+    usr = LibGit2.getconfig("github.user", "")
+    if isempty(usr)
+        throw(PkgError("""
         no GitHub user name configured; please configure it with:
 
             git config --global github.user USERNAME
 
         where USERNAME is replaced with your GitHub user name.
-        """)
+        """))
     end
-    readchomp(`git config --global github.user`)
+    return usr
 end
 
 function json()
     isdefined(:JSON) || try require("JSON")
     catch err
         warn(err)
-        error("using the GitHub API requires having the JSON package installed ")
+        throw(PkgError("using the GitHub API requires having the JSON package installed "))
     end
     Main.JSON
 end
 
 function curl(url::AbstractString, opts::Cmd=``)
-    success(`curl --version`) || error("using the GitHub API requires having `curl` installed")
+    success(`curl --version`) || throw(PkgError("using the GitHub API requires having `curl` installed"))
```

Comment:
No, this file needs to do a bunch more things like posting that `URLDownloadToFileW` can't do. Base's download is also kind of crippled and broken in a few ways like not following redirects correctly on Windows, ref https://github.com/JuliaLang/BinDeps.jl/issues/135. We could ship curl, but it would need some ca certificates and mozilla-nss and other nonsense to work correctly over https. `Invoke-RestMethod` in powershell is quite nice actually, having an object-oriented shell that understands json natively is pretty useful.